hydra/_internal/utils.py

```diff
@@ -402,6 +402,12 @@ def _get_completion_help() -> str:
             completion_info.append(head)
             completion_info.append(plugin_cls.help(cmd).format(_get_exec_command()))
         completion_info.append("")
+
+    completion_info.append(
+        "Zsh is compatible with the Bash shell completion, see the [documentation]"
+        "(https://hydra.cc/docs/tutorials/basic/running_your_app/tab_completion#zsh-instructions)"
+        " for details."
+    )
```

Comment:
Can you standardize here?

You can create:
```python
class ZshCompletion(BashCompletion):
   ...
```

With the appropriate functions (help()? _get_exec_commands()?)

Reply:
Makes sense.

However, when trying to inherit from BashCompletion I get the following error:
```
Traceback (most recent call last):
  File "/home/selflein/Developer/git/hydra/hydra/_internal/utils.py", line 196, in run_and_report
    return func()
  File "/home/selflein/Developer/git/hydra/hydra/_internal/utils.py", line 370, in <lambda>
    lambda: hydra.shell_completion(
  File "/home/selflein/Developer/git/hydra/hydra/_internal/hydra.py", line 229, in shell_completion
    shell_to_plugin = self.get_shell_to_plugin_map(self.config_loader)
  File "/home/selflein/Developer/git/hydra/hydra/_internal/hydra.py", line 216, in get_shell_to_plugin_map
    raise ValueError(f"Multiple plugins installed for {shell} : {lst}")
ValueError: Multiple plugins installed for bash : BashCompletion,BashCompletion
```

The issue seems to be within Plugin discovery since [here](https://github.com/facebookresearch/hydra/blob/8adcee0f61ab61dc6cd85dd046ab568c1bd102a0/hydra/core/plugins.py#L193) for `ZshCompletion` the parent class `BashCompletion` is included as well. Any idea for a workaround?

Reply:
Oh, ouch.
I guess you can implement it as composition instead (delegate to a nested bash completion plugin instead of be one).

Reply:
Okay I tried something along these lines:
```
class ZshCompletion(CompletionPlugin):
    def __init__(self, config_loader):
        super().__init__(config_loader)
        self.delegate = BashCompletion(config_loader)
```
However, this will still result in the same error as above.

Reply:
Can you update the diff? I want to see what you tried exactly.

Reply:
Updated the diff.